src/include/ikarus/Grids/GridEntities/DefaultGridEntities.h

```diff
@@ -61,96 +66,62 @@ namespace Ikarus::Grid {
-      if constexpr (mydimension == 1)
-        return getChildVertices().size();
-      else if constexpr (mydimension == 2) {
-        if (codim == 1)
-          return getChildEntities<1>().size();
-        else if (codim == 2)
-          return getChildEntities<0>().size();
-        else  // codim = 3
-          throw std::logic_error("A entity with dimension 2 does not have subentities with codim 3!");
-      } else {  //(mydimension == 3)
-        if (codim == 1)
-          return getChildEntities<2>().size();
-        else if (codim == 2)
-          return getChildEntities<1>().size();
-        else  // codim = 3
-          return getChildEntities<0>().size();
-      }
-    }
+    [[nodiscard]] unsigned int subEntities(unsigned int codim) const;
+
     /** \brief Return the fundamental geometric type of the entity */
-    Dune::GeometryType type() const;
+    [[nodiscard]] Dune::GeometryType type() const;
 
     /** \brief Returns the geometric realization of the entity */
-    auto geometry() const {
-      std::vector<Dune::FieldVector<double, dimensionworld>> fieldVectorVector;
-      for (const auto& pos :
-           std::ranges::transform_view(getChildVertices(), &DefaultGridEntity<griddim, griddim, wdim>::getPosition))
-        fieldVectorVector.push_back(toFieldVector(pos));
-
-      return Ikarus::Geometry::IGeometry(Geometry(type(), fieldVectorVector));
-    }
+    auto geometry() const;
 
     /** \brief Return copy of the id of this entity */
     [[nodiscard]] size_t getID() const { return id; }
 
     /** \brief Get refínement level where this entity belongs to*/
     [[nodiscard]] int level() const { return levelIndex; }
 
+  private:
     /** \brief The refinement level to which this entity belongs */
     int levelIndex{};
-
-    /** \brief The index of this element on the leaf level */
-    int leafIndex{};
-
-  private:
     /** \brief A persistent id of this entity*/
     size_t id{};
 
     /** \brief Childrens of the entity on the current grid , i.e. surfaces of a cube*/
     EntitiesChildernType entitiesChildren;
 
-    /** \brief Childrens of the entity on a finer grid , i.e. subcubes of a cube*/
+    /** \brief Childrens of the entity on a finer grid , i.e. subcubes of a cube, it stores only pointers to the
+     * entities.*/
     std::vector<DefaultGridEntity<dimension, 0, dimensionworld>*> levelEntitiesChildren;
   };
 
   /**
    * \brief DefaultGridEntity
    *
-   * \param  griddim  The
+   * \tparam  griddim  The dimension of the grid
+   * \tparam  cogriddim=griddim  The codimension of the entity. Here
+   * \tparam  wdim  The dimension of the world space where the grid is embedded
    *
    * \note Partial template specialization for entities with codim == griddim
    * These entities have no grid children, since they are vertices!
    **/
   template <int griddim, int wdim>
   class DefaultGridEntity<griddim, griddim, wdim> {
   public:
+    static constexpr int dimension      = griddim;

```

Comment:
I removed some redundant comments. Should i also remove some comments. E.g. for levelIndex? 
This increases readability and the grid Entity class will maybe fit one one screen. @ToWibb What do you think?
Should comments only be applied to strange and difficult typedefs? See below examples.